components/datetime/src/options/components.rs

```diff
@@ -362,16 +365,86 @@ pub enum Month {
     Narrow,
 }
 
+// Each enum variant is documented with the UTS 35 field information from:
+// https://unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table
+//
 /// Options for displaying a time zone for the `components::`[`Bag`].
+///
+/// Note that the initial implementation is focusing on only supporting ECMA-402 compatible
+/// options.
 #[derive(Debug, Clone, Copy, PartialEq)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum TimeZoneName {
-    #[allow(missing_docs)] // TODO(#686) - Add missing docs.
-    #[cfg_attr(feature = "serde", serde(rename = "long"))]
-    Long,
-    #[allow(missing_docs)] // TODO(#686) - Add missing docs.
+    // UTS-35 fields: z..zzz
+    //
+    /// Short localized form, without the location. (e.g.: PST, GMT-8)
     #[cfg_attr(feature = "serde", serde(rename = "short"))]
     Short,
+
+    // UTS-35 fields: zzzz
+    // Per UTS-35: [long form] specific non-location (falling back to long localized GMT)
+    //
+    /// Long localized form, without the location (e.g., Pacific Standard Time, Nordamerikanische Westküsten-Normalzeit)
+    #[cfg_attr(feature = "serde", serde(rename = "long"))]
+    Long,
```

Comment:
> suggestion

I think it would be best to call these `ShortSpecific` and `LongSpecific`, which are the opposites of `ShortGeneric` and `LongGeneric`. 

They're all non-location formats, but the `Specific` ones account for variants such as daylight savings time where as the `Generic` ones do not.